docs/design/network-policy-support.md

```diff
@@ -0,0 +1,246 @@
+
+## Multicluster Network Policy support
+
+This document describes the Network Policy use cases that we propose
+supporting for a first iteration in Submariner. In this iteration, the
+intention is to support the ingress side of existing k8s Network Policies
+as is, while in the future we could explore with the kubernetes community
+options to extend Network Policies for multi cluster scenarios.
+
+> A Kubernetes network policy is a specification of how pods are allowed
+> to communicate with each other and other network endpoints. 
+> It is similar to Access Control Lists where one can specify the traffic
+> that would be allowed/denied to the pods. Network Policies are additive,
+> so having two Network Policies that select the same pods will result in
+> allowing both defined policies.
+
+
+> Network Policies are implemented by the underlying networking CNI plugin
+> for the k8s cluster. Some CNIs don't implement Network Policies, this
+> feature assumes that the CNI being used at least provides intra-cluster
+> Network policy support.
+
+## Diagram
+
+The following diagram depicts two clusters and two namespaces. Those resources
+will be referenced in the use cases.
+
+![Diagram](img/network-policies-base-diagram.png)
+
+The cluster labels for the namespaces are just illustrative examples of
+label assignment. The admin could follow that pattern or a different
+pattern. In future, we could integrate with federation
+[ClusterID](https://docs.google.com/document/d/1F__vEKeI41P7PPUCMM9PVPYY34pyrvQI5rbTJVnS5c4/edit)
+which is being discussed in the k8s multicluster-sig.
+
+For the initial implementation (still under design) it is assumed that
+pod/service CIDRs are non-overlapping, and that we preserve the
+source IP address on the datapath. In parallel we are also considering
+technical solutions to be able to support overlapping CIDRs in the future.
+
+The examples shown below can be extrapolated to UDP/SCTP too.
+
+We are only trying to tackle Ingress policies for a first iteration, same
+design could apply to egress policy for later iterations.
+
+## Use cases
+
+
+
+| Use-case |       Description                                       |
+| -------  | ------------------------------------------------------- |
+|    1     | Connected Clusters without any Network Policy           |
+|    2     | Allow frontend pods from a remote cluster in the same project to access the backend pods on the local cluster. |
```

Comment:
What is a project?  That's not a kubernetes term of art?  It's not clearly a namespace, because a namespace subdivides a cluster, and you said "remote cluster in the same project".  So I can't evaluate this very well.

Reply:
Right, the correct term here would be namespace.